### PR TITLE
fix(authentication): don't crash if an error occurs when initializing the authentication client

### PR DESCRIPTION
**Pull request by ghusse**
Linked to CU-860rxhayg

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/forestadmin-client/src/auth/index.ts [hunk 1]
```diff
@@ -56,7 +75,7 @@ export default class AuthService implements ForestAdminAuthServiceInterface {
     scope: string;
     state: string;
   }): Promise<string> {
-    if (!this.client) throw new Error('AuthService not initialized');
+    await this.createClient();
 
     const url = this.client.authorizationUrl({
```
Comment (Scra3) [inline packages/forestadmin-client/src/auth/index.ts:80]: If the createClient fails, this.client is undefined

Comment (ghusse) [reply packages/forestadmin-client/src/auth/index.ts:80]: If `createClient` fails, it throws an error here. The `try/catch` is only in the init method.

File: packages/forestadmin-client/src/auth/index.ts [hunk 2]
```diff
@@ -12,13 +12,32 @@ export default class AuthService implements ForestAdminAuthServiceInterface {
...
+      // server or network error. We don't want the application to crash.
+      this.options.logger(
+        'Warn',
+        // eslint-disable-next-line max-len
+        `Error while registering the authentication client. Authentication might not work: ${e.message}`,
+      );
+    }
+  }
+
+  protected async createClient() {
```
Comment (Scra3) [inline packages/forestadmin-client/src/auth/index.ts:33]: protected method should be after the public method

File: packages/forestadmin-client/src/auth/index.ts (hunk 1 above)
Comment (Scra3) [inline packages/forestadmin-client/src/auth/index.ts:80]: optional: you can return here

File: packages/forestadmin-client/src/auth/index.ts (hunk 2 above)
Comment (ghusse) [reply packages/forestadmin-client/src/auth/index.ts:33]: If it's a rule, it should be enforced by eslint.

File: packages/forestadmin-client/src/auth/index.ts (hunk 1 above)
Comment (ghusse) [reply packages/forestadmin-client/src/auth/index.ts:80]: Return what ?

**Review by Scra3 — APPROVED** [review]

**Pull request merged**